frontend/src/main/java/com/google/tripmeout/frontend/servlets/PlaceIndividualServlet.java

```diff
@@ -0,0 +1,154 @@
+                                     .setPlaceId(placeId)
+                                     .setName(details.name)
+                                     .setLatitude(details.geometry.location.lat)
+                                     .setLongitude(details.geometry.location.lng)
+                                     .build();
+
+      placeStorage.updateUserMarkOrAddPlaceVisit(newPlace, status);
+      response.setStatus(HttpServletResponse.SC_OK);
+
+    } catch (IllegalArgumentException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+
+    } catch (InvalidRequestException e) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+
+    } catch (Exception e) {
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    try {
+      Matcher matcher = ServletUtil.matchUriOrThrowError(request, URI_NAME_PATTERN);
+      String tripId = matcher.group(1);
+      String placeId = matcher.group(2);
+
+      placeStorage.removePlaceVisit(tripId, placeId);
+      response.setStatus(HttpServletResponse.SC_OK);
+
+    } catch (IllegalArgumentException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+
+    } catch (PlaceVisitNotFoundException e) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+
+    } catch (Exception e) {
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      Matcher matcher = ServletUtil.matchUriOrThrowError(request, URI_NAME_PATTERN);
+      String tripId = matcher.group(1);
+      String placeId = matcher.group(2);
+
+      Optional<PlaceVisitModel> optionalPlace = placeStorage.getPlaceVisit(tripId, placeId);
+
+      String userMark;
+
+      if (optionalPlace.isPresent()) {
+        userMark = optionalPlace.get().userMark().toString();
+      } else {
+        userMark = PlaceVisitModel.UserMark.UNKNOWN.toString();
+      }
+
+      PlaceDetailsRequest place = new PlaceDetailsRequest(CONTEXT).placeId(placeId).fields(
+          PlaceDetailsRequest.FieldMask.NAME, PlaceDetailsRequest.FieldMask.BUSINESS_STATUS,
+          PlaceDetailsRequest.FieldMask.FORMATTED_ADDRESS,
+          PlaceDetailsRequest.FieldMask.FORMATTED_PHONE_NUMBER,
+          PlaceDetailsRequest.FieldMask.GEOMETRY_LOCATION,
+          PlaceDetailsRequest.FieldMask.OPENING_HOURS, PlaceDetailsRequest.FieldMask.PHOTOS,
+          PlaceDetailsRequest.FieldMask.PRICE_LEVEL, PlaceDetailsRequest.FieldMask.RATING,
+          PlaceDetailsRequest.FieldMask.REVIEW, PlaceDetailsRequest.FieldMask.TYPES,
+          PlaceDetailsRequest.FieldMask.WEBSITE);
+      PlaceDetails details = place.await();
+
+      String[] newHtmlAttributions = new String[details.htmlAttributions.length + 1];
```

Comment:
Avoid using arrays wherever possible. This should be a `List<String>` or similar.